src/storage.js

```diff
@@ -0,0 +1,81 @@
+"use strict";
+
+const fs = require("fs");
+const fsextra = require("fs-extra");
+const path = require("path");
+const crypto = require("crypto");
+const helper = require("./helper");
+
+class Storage {
+	constructor() {
+		this.references = new Map();
+
+		// Ensures that a directory is empty.
+		// Deletes directory contents if the directory is not empty.
+		// If the directory does not exist, it is created.
+		fsextra.emptyDirSync(helper.getStoragePath());
```

Comment:
I'm still unsure about clearing at startup. More importantly, a FS could still be filled until server gets restarted. I know you have that garbage collector when no references exist anymore, but that still leaves room for fs problems.